base_layer/core/src/proof_of_work/difficulty.rs

```diff
@@ -101,6 +80,14 @@ impl From<u64> for Difficulty {
     }
 }
 
+pub trait DifficultyAjustment {
```

Comment:
I wonder if there shouldn't be a way to rewind the added samples (timestamp, cum_difficulty) as these samples can get reorged and should be removed and replaced with the new samples from the longest chain.

Reply:
Good point, but since re-orgs happen very seldom, it's not so bad to repopulate the vector from scratch when they do happen